src/peergos/server/tests/simulation/Simulator.java

```diff
@@ -486,44 +558,79 @@ private boolean verifySharingPermissions(String user, Path path) {
+                    switch (permission) {
+                        case READ:
+                            try {
+                                //can read?
+                                fs.ls(path);
+                            } catch (Exception ex) {
+                                LOG.log(Level.SEVERE, "User " + sharee + " could not read shared-path " + path + "!", ex);
+                                isVerified = false;
+                            }
+                            break;
+                        case WRITE:
+                            try {
+                                //can read?
+                                fs.ls(path);
+                            } catch (Exception ex) {
+                                LOG.log(Level.SEVERE, "User " + sharee + " could not read shared-path " + path + "!", ex);
+                                isVerified = false;
+                            }
+                            if (isVerified) {
+                                try {
+                                    mkdir(user, path.resolve(getNextName()));
+                                } catch (Exception ex) {
+                                    LOG.log(Level.SEVERE, "User " + sharee + " could not write  shared-path " + path + "!", ex);
+                                    isVerified = false;
+                                }
+                            }
+                            break;
+                        default:
+                            throw new IllegalStateException();
+                    }
+                } else {
+                    switch (permission) {
+                        case READ:
+                            try {
+                                //can read?
+                                read = fs.read(path);
+                            } catch (Exception ex) {
+                                LOG.log(Level.SEVERE, "User " + sharee + " could not read shared-path " + path + "!", ex);
+                                isVerified = false;
+                            }
+                            break;
+                        case WRITE:
                             try {
-                                // can overwrite?
-                                fs.write(path, new byte[]{read[0]});
+                                //can read?
+                                read = fs.read(path);
                             } catch (Exception ex) {
-                                LOG.log(Level.SEVERE, "User " + sharee + " could not write  shared-path " + path + "!", ex);
+                                LOG.log(Level.SEVERE, "User " + sharee + " could not read shared-path " + path + "!", ex);
                                 isVerified = false;
                             }
-                        }
-                        break;
-                    default:
-                        throw new IllegalStateException();
+                            if (isVerified) {
```

Comment:
Why only do  this if  the permissions are  correct? Also this way it won't log the problem (that the path is  not writable).

Reply:
every time isVerfied is set to false a SEVERE log message is written. If that occurs, i expect the bug to be fixed and then the extra checks are done.